### DISCUSSION TITLE
Settings in dotnet Plugin

### DISCUSSION THREAD

**Discussion by HorridModz** [Flow-Launcher/Flow.Launcher#2267] [Q&A]
Hi, I'm writing a FlowLauncher plugin in C#. My plugin is going to have several settings. I have scoured the documentation and example plugins, but I have not figured out how to implement settings. It appears to be different between languages - I was able to figure out how to do it for python plugins, but not for dotnet plugins.

Can I please have some pointers on how to implement settings?

**Comment by jjw24** [comment]
Hi there, the way to create .Net and non-.Net plugin settings panel is different.

You need to inherit `ISettingsProvider`, which will then require you to implement `public Control CreateSettingPanel()`. Follow the Sys plugin as an example https://github.com/Flow-Launcher/Flow.Launcher/blob/1e16c5fdbef23e794722f760d650d0ae2a036698/Plugins/Flow.Launcher.Plugin.Sys/Main.cs#L50.

You can also have look at some of flow's default plugins for more complex examples.
